cdk_infra/lib/stacks/msk/msk-stack.ts

```diff
@@ -0,0 +1,141 @@
+      serverProperties: SERVER_PROPERTIES,
+      kafkaVersionsList: MSK_VERSIONS
+    });
+
+    const securityGroupMSKAOC = this.createSecurityGroup(
+      'aoc-msk',
+      props.aocVPC
+    );
+    const securityGroupMSKEKS = this.createSecurityGroup(
+      'eks-msk',
+      props.eksVPC
+    );
+
+    MSK_VERSIONS.forEach((version) =>
+      this.createCluster(
+        'AOCMSKCluster',
+        version,
+        configuration,
+        props.aocVPC,
+        securityGroupMSKAOC
+      )
+    );
+    MSK_VERSIONS.forEach((version) =>
+      this.createCluster(
+        'EKSMSKCluster',
+        version,
+        configuration,
+        props.eksVPC,
+        securityGroupMSKEKS
+      )
+    );
+  }
+
+  private createSecurityGroup(sgName: string, vpc: aws_ec2.IVpc) {
+    const securityGroupMSKAOC = new SecurityGroup(this, `sg-${sgName}`, {
+      vpc: vpc,
+      securityGroupName: sgName
+    });
+    // plain text listener
+    securityGroupMSKAOC.addIngressRule(
+      Peer.ipv4(vpc.vpcCidrBlock),
+      Port.tcp(9092)
+    );
+    // TLS listener
+    securityGroupMSKAOC.addIngressRule(
+      Peer.ipv4(vpc.vpcCidrBlock),
+      Port.tcp(9094)
+    );
+
+    return securityGroupMSKAOC;
+  }
+
+  private createCluster(
+    clusterPrefix: string,
+    version: string,
+    configuration: msk.CfnConfiguration,
+    vpc: IVpc,
+    securityGroup: aws_ec2.SecurityGroup
+  ): msk.CfnCluster {
+    const clusterName = `${clusterPrefix}${version.replaceAll('.', '-')}`;
+    const logGroupName = `/aws-otel-tests/${clusterName}`;
+    const logGroup = new LogGroup(this, `loggroup-${clusterName}`, {
+      logGroupName: logGroupName,
+      retention: RetentionDays.ONE_MONTH,
+      removalPolicy: RemovalPolicy.DESTROY
+    });
+
+    return new msk.CfnCluster(this, `cluster-${clusterName}`, {
+      clusterName: clusterName,
+      kafkaVersion: version,
+      numberOfBrokerNodes: 3,
+      configurationInfo: {
+        arn: configuration.attrArn,
+        revision: 1
+      },
+      brokerNodeGroupInfo: {
+        instanceType: 'kafka.t3.small',
+        clientSubnets: vpc.privateSubnets.map((x) => x.subnetId),
+        storageInfo: {
+          ebsStorageInfo: {
+            volumeSize: 10
+          }
+        },
+        securityGroups: [securityGroup.securityGroupId]
+      },
+      clientAuthentication: {
+        // No need to authenticate since we are only accessible inside the vpc
+        unauthenticated: {
+          enabled: true
+        }
+      },
+      encryptionInfo: {
+        encryptionInTransit: {
+          clientBroker: 'TLS_PLAINTEXT'
+        }
+      },
+      loggingInfo: {
+        brokerLogs: {
+          cloudWatchLogs: {
+            enabled: true,
+            logGroup: logGroup.logGroupName
+          }
+        }
+      }
+    });
+  }
+}
+
+export interface MSKStackProps extends StackProps {
+  eksVPC: IVpc; // VPC used in the EKS Clusters
+  aocVPC: IVpc; // Legacy VPC used in the remaining in the tests
```

Comment:
Can we instead accept a set of VPCs as input and build clusters based on that?

Reply:
Instead of making the stack generic, I made the setup for generating a msk cluster for tests generic: i create the `MSKTestCluster` construct. 

The reason why we cannot make the stack generic is because we need to lookup for the aoc VPC that was created in terraform. the lookup can only happen inside a stack, therefore it is not possible to simple pass a set of vpcs to the msk stack. One alternative would be to create a stack just to perform the lookup, but I think that would add too much complexity for little benefit.